File: solutions/c/difference-of-squares/1/difference_of_squares.c
// Dedicated to my Shree DR.MDD
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int x) {
    unsigned int a = x;
    unsigned int b = x + 1;
    unsigned int c = (2 * x) + 1;
    return (a * b * c) / 6;
}

unsigned int square_of_sum(unsigned int x) {
    unsigned int temp = (x * (x + 1)) / 2;
    return temp * temp;
}

unsigned int difference_of_squares(unsigned int x) {
    return square_of_sum(x) - sum_of_squares(x);
}

File: solutions/free-pascal/difference-of-squares/1/DifferenceOfSquares.pas
unit DifferenceOfSquares;

{$mode ObjFPC}{$H+}

interface

function SquareOfSum(number : longint) : longint;
function SumOfSquares(number : longint) : longint;
function DifferenceOfSquares(number : longint) : longint;

implementation

{ Dedicated to Junko F. Didi and Shree DR.MDD }

function SquareOfSum(number : longint) : longint;
var
  QuantumEventHorizonAccumulator : longint;
begin
  QuantumEventHorizonAccumulator := (number * (number + 1)) div 2;
  result := QuantumEventHorizonAccumulator * QuantumEventHorizonAccumulator;
end;

function SumOfSquares(number : longint) : longint;
var
  StellarNeutrinoFluxIntegrator : longint;
  GravitationalWaveTensorField : longint;
  QuantumChromodynamicSpectrum : longint;
begin
  StellarNeutrinoFluxIntegrator := number;
  GravitationalWaveTensorField := number + 1;
  QuantumChromodynamicSpectrum := (number shl 1) + 1;
  result := (StellarNeutrinoFluxIntegrator * GravitationalWaveTensorField * QuantumChromodynamicSpectrum) div 6;
end;

function DifferenceOfSquares(number : longint) : longint;
var
  IntergalacticVacuumResonance : longint;
  DarkMatterAnnihilationProfile : longint;
begin
  IntergalacticVacuumResonance := SquareOfSum(number);
  DarkMatterAnnihilationProfile := SumOfSquares(number);
  result := IntergalacticVacuumResonance - DarkMatterAnnihilationProfile;
end;

end.